### ISSUE TITLE
setScaleLimit does not work with setScaleToFit(false)

### ISSUE THREAD

**Issue by reachingyu** [moagrius/TileView#174]
I am trying to set the minimum scale for my map using setScaleLimit and setScaleToFit. I have read previous closed issues on setScaleToFit and how setScaleLimit should be respected if setScaleToFit is false. However, when I pinch zoom out on my map, it can zoom out all the way to 0 (disappear) without respecting the minimum limits.  Why is this? and how can I prevent this from happening?

Here is my code: 

{
    mInflater = inflater;
    mView = inflater.inflate(R.layout.fragment_match, container);

    mTileView = new TileView(getActivity());
    mTileView.setDecoder(new BitmapDecoderAssets());

    mTileView.setSize(4096, 3813);

    mTileView.addDetailLevel(0.125f, "tiles/map/125_%col%_%row%.png", "samples/map_real_android.png");
    mTileView.addDetailLevel(0.25f, "tiles/map/250_%col%_%row%.png", "samples/map_real_android.png");
    mTileView.addDetailLevel(0.5f, "tiles/map/500_%col%_%row%.png", "samples/map_real_android.png");
    mTileView.addDetailLevel(1f, "tiles/map/1000_%col%_%row%.png", "samples/map_real_android.png");

    mTileView.setScale(0.49);
    mTileView.setScaleLimits(0.3, 0.49);
    mTileView.setScaleToFit(false);
    mTileView.setId(R.id.map);

    RelativeLayout.LayoutParams lp = new RelativeLayout.LayoutParams(
        RelativeLayout.LayoutParams.WRAP_CONTENT, RelativeLayout.LayoutParams.WRAP_CONTENT);
    ((RelativeLayout) mView).addView(mTileView, lp);

    return mView;
 }

My map in full screen:
![screen shot 2015-05-13 at 10 56 21 pm](https://cloud.githubusercontent.com/assets/3440261/7626787/60309018-f9c3-11e4-8b07-aae04ab1f9a5.png)

My map when zoomed out:
![screen shot 2015-05-13 at 10 59 36 pm](https://cloud.githubusercontent.com/assets/3440261/7626853/66a87a18-f9c4-11e4-8354-0a995488c3bd.png)

My (desired) zoom out size/scale:
![screen shot 2015-05-13 at 11 01 13 pm](https://cloud.githubusercontent.com/assets/3440261/7626856/78a805d0-f9c4-11e4-9fd3-9eb64689f8f0.png)

Thank you!

**Comment by moagrius** [comment]
At the moment you're setting the minimum scale, the instance is set to scale to fit, so the minimum directive won't be applied.  Try changing the order, like so:

```
mTileView.setScaleToFit(false);
mTileView.setScaleLimits(0.3, 0.49);
mTileView.setScale(0.49);
```

Post back if that doesn't work.

I'm going to leave this issue open since this might be something I'll "fix" in the next update.

**Comment by reachingyu** [comment]
Thank you for the explanation! Your solution works for me.

**Comment by moagrius** [comment]
Glad to hear it - thanks for posting

**Comment by moagrius** [comment]
This has been corrected in version 2.0.0
